File: examples/ts/p256-primitives/P256Primitives.test.ts
import { describe, it, expect } from 'vitest';
import { readFileSync } from 'node:fs';
import { join, dirname } from 'node:path';
import { fileURLToPath } from 'node:url';
import { TestContract } from 'runar-testing';

const __dirname = dirname(fileURLToPath(import.meta.url));
const source = readFileSync(join(__dirname, 'P256Primitives.runar.ts'), 'utf8');

// P256Primitives.runar.ts exercises NIST P-256 builtins (p256Mul, p256Add,
// p256MulGen). Constructing real P-256 points from native TS duplicates work
// covered by the runar-sdk tier — peer Zig/Go suites also keep this fixture
// at frontend coverage. TestContract.fromSource itself is the compile-check
// (parse → validate → typecheck → ANF → stack → emit).
const ZERO_PT_HEX = '00'.repeat(64);

describe('P256Primitives', () => {
  it('compiles via TestContract.fromSource', () => {
    const c = TestContract.fromSource(source, { expectedPoint: ZERO_PT_HEX });
    expect(c.state.expectedPoint).toBe(ZERO_PT_HEX);
  });
});

File: examples/zig/p256-primitives/P256Primitives_test.zig
const std = @import("std");
const root = @import("../examples_test.zig");

// P256Primitives.runar.zig exercises NIST P-256 builtins (`p256Mul`,
// `p256Add`, `p256MulGen`). Constructing real P-256 points from native Zig
// duplicates work that lives in the runar-zig SDK tier, so for this fixture
// we focus on Rúnar frontend coverage (parse → validate → typecheck).

fn contractPath(comptime basename: []const u8) []const u8 {
    return "p256-primitives/" ++ basename;
}

test "compile-check P256Primitives.runar.zig (source)" {
    try root.runar.compileCheckSource(
        std.testing.allocator,
        @embedFile("P256Primitives.runar.zig"),
        "P256Primitives.runar.zig",
    );
}

test "compile-check P256Primitives.runar.zig (file)" {
    try root.runar.compileCheckFile(
        std.testing.allocator,
        contractPath("P256Primitives.runar.zig"),
    );
}
